src/app/access-requests/services/access-request.service.ts

```diff
@@ -285,6 +290,108 @@ export class AccessRequestService {
       .patch<null>(`${this.#arsEndpointUrl}/${id}`, changes)
       .pipe(tap(() => this.#updateAccessRequestLocally(id, changes)));
   }
+
+  // Similar structure to what we do for access requests but for access grants
+  #allAccessGrantsFilter = signal<AccessGrantFilter | undefined>(undefined);
+  allAccessGrantsFilter = computed(
+    () =>
+      this.#allAccessGrantsFilter() ?? {
+        status: undefined,
+        name: undefined,
+        email: undefined,
+        dataset_id: undefined,
+        grant_id: undefined,
+      },
+  );
+
+  /**
+   * Load all access grants
+   * @param filter the filter to apply
+   */
+  setAllAccessGrantsFilter(filter: AccessGrantFilter): void {
+    this.#allAccessGrantsFilter.set(
+      filter.name ||
+        filter.status ||
+        filter.dataset_id ||
+        filter.email ||
+        filter.grant_id
+        ? filter
+        : undefined,
+    );
```

Comment:
```suggestion
    this.#allAccessGrantsFilter.set(filter);
```

Did that for the other filters too. Otherwise you need to carefully match the condition with the default value for the filter (e,g. if we want to use a default status value).